stubs/nvml_stubs.c

```diff
@@ -574,47 +552,36 @@ stub_nvml_device_get_active_vgpus(value ml_interface, value ml_device)
 
     interface = (nvmlInterface *) ml_interface;
     device = *(nvmlDevice_t *) ml_device;
+    list = Val_emptylist;
 
-    // Get instances dynamically increasing the buffer size
-    int dummy;
-    do {
-        error = interface->deviceGetActiveVgpus(device,
-                                                &vgpuCount,
-                                                (vgpuInstances) ?
-                                                vgpuInstances
-                                                :
-                                                (nvmlVgpuInstance_t
-                                                 *) & dummy);
-        if ((error == NVML_ERROR_INSUFFICIENT_SIZE)
-            && (vgpuCount > 0)) {
-            if (vgpuInstances) {
-                free(vgpuInstances);
-            }
-            vgpuInstances = (nvmlVgpuInstance_t *)
-                malloc(sizeof(nvmlVgpuInstance_t) * vgpuCount);
-            if (!vgpuInstances) {
-                check_error(interface, NVML_ERROR_MEMORY);
-            }
-        }
+    error =
+        interface->deviceGetActiveVgpus(device, &vgpuCount, vgpuInstances);
+    if (error == NVML_SUCCESS) {
+        CAMLreturn(list);       /* no active vGPU */
```

Comment:
so this'd be an assertion that vgpucount == 0, maybe add a comment that we'd expect NVML_ERROR_INSUFFICIENT_SIZE if it was anything > 0.